src/main.rs

```diff
@@ -57,3 +61,21 @@ fn main() -> Result<(), Box<dyn error::Error>> {
     info!("Duration: {}", format_duration(start.elapsed()).to_string());
     Ok(())
 }
+
+fn read_input(input_path: &path::PathBuf) -> Result<SetOntology<RcStr>, Box<dyn error::Error>> {
+    let file = File::open(&input_path)?;
+    let mut bufreader = BufReader::new(file);
+
+    let config = ParserConfiguration::default();
+    match input_path.extension().and_then(|s| s.to_str()) {
+        Some("owx") => {
+            let ret = horned_owl::io::owx::reader::read(&mut bufreader, config).expect("unable to parse input");
+            Ok(ret.0)
+        }
+        Some("owl") => {
+            let ret = horned_owl::io::rdf::reader::read(&mut bufreader, config).expect("unable to parse input");
+            Ok(ret.0.into())
+        }
+        _ => Err(Box::<dyn error::Error>::from("unable to parse input")),
```

Comment:
Why not return the error from horned-owl? Could it provide useful information that could be logged about why parsing failed?

Reply:
The leafs of the match are different.  I will change the .expect() to ? so that any error will trickle up.

Reply:
Okay, thanks.